ggml/src/ggml-cuda/mmq.cuh

```diff
@@ -1268,23 +1680,23 @@ template <int mmq_y, int nwarps, bool need_check> static __device__ __forceinlin
 
         const int sc = __vsubss4(sc_low | sc_high, 0x20202020);
 
-#ifdef NEW_MMA_AVAILABLE
+#if defined(AMD_MMA_AVAILABLE) || defined(NEW_MMA_AVAILABLE)
         const int8_t * sc8 = (const int8_t *) &sc;
         const float d = bxi->d;
 
 #pragma unroll
         for (int l = 0; l < int(sizeof(int)); ++l) {
-            x_df[i*MMQ_MMA_TILE_X_K_Q3_K + sizeof(int)*(threadIdx.x % (WARP_SIZE/8)) + l] = d*sc8[l];
+            x_df[i*MMQ_MMA_TILE_X_K_Q3_K + sizeof(int)*ksc + l] = d*sc8[l];
         }
 #else
-        x_sc[i*(WARP_SIZE/8) + i/8 + threadIdx.x % (WARP_SIZE/8)] = sc;
+        x_sc[i*4 + i/8 + ksc] = sc;
```

Comment:
```suggestion
        x_sc[i*(MMQ_TILE_NE_K/8) + i/8 + ksc] = sc;
```

I think some semantic meaning is lost here. The reason for the hard-coded value 8 is that for every 8 rows with `WARP_SIZE` quant values there are `WARP_SIZE` scale values so you need 1 value of padding per 8 rows.